compilation error in DoublyLinkedList with Java 21

```
 * JGraphT version: 7084d919351ed09f36a82230fd6f47c8e805d65e
 * Java version (java -version)/platform:  21-ea+22-1890
```

**Issue**

JGraphT does not compile with the latest versions of Java, due to the implementation of [sequenced collections](https://openjdk.org/jeps/431).

Example failure:

```
[ERROR] ./jgrapht-core/src/main/java/org/jgrapht/util/DoublyLinkedList.java:[59,8] types java.util.Deque<E> and java.util.List<E> are incompatible;
  both define reversed(), but with unrelated return types
```

**Steps to reproduce (small coding example)**

```diff
$ git diff
diff --git a/pom.xml b/pom.xml
index 1d82de46df..0c681446f4 100644
--- a/pom.xml
+++ b/pom.xml
@@ -232,7 +232,7 @@
                     <artifactId>maven-compiler-plugin</artifactId>
                     <version>${maven-compiler-plugin.version}</version>
                     <configuration>
-                        <release>11</release>
+                        <release>21</release>
                         <compilerArgs>
                             <arg>--add-modules</arg>
                                 <arg>java.xml</arg>
```

```
$ cd jgrapht-core
$ JAVA_HOME=${JDK21_HOME?} mvn clean package -DskipTests=true
```

**Other information**

* https://openjdk.org/jeps/431
* https://github.com/openjdk/jdk/commit/17ce0976e442d5fabb14daed40fa9a768989f02e

--- a/jgrapht-core/src/main/java/org/jgrapht/util/DoublyLinkedList.java
+++ b/jgrapht-core/src/main/java/org/jgrapht/util/DoublyLinkedList.java
@@ -58,7 +58,6 @@ import java.util.function.*;
  */
 public class DoublyLinkedList<E>
     extends AbstractSequentialList<E>
-    implements Deque<E>
 {
     /** The first element of the list, {@code null} if this list is empty. */
     private ListNodeImpl<E> head = null;
@@ -652,48 +651,28 @@ public class DoublyLinkedList<E>
 
     // Deque methods
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public void addFirst(E e)
     {
         addElementFirst(e);
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public void addLast(E e)
     {
         addElementLast(e);
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public boolean offerFirst(E e)
     {
         addElementFirst(e);
         return true;
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public boolean offerLast(E e)
     {
         addElementLast(e);
         return true;
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public E removeFirst()
     {
         if (isEmpty()) {
@@ -705,10 +684,6 @@ public class DoublyLinkedList<E>
         return node.getValue();
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public E removeLast()
     {
         if (isEmpty()) {
@@ -720,10 +695,6 @@ public class DoublyLinkedList<E>
         return node.getValue();
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public E pollFirst()
     {
         if (isEmpty()) {
@@ -734,10 +705,6 @@ public class DoublyLinkedList<E>
         return node.getValue();
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public E pollLast()
     {
         if (isEmpty()) {
@@ -748,46 +715,26 @@ public class DoublyLinkedList<E>
         return node.getValue();
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public E getFirst()
     {
         return getFirstNode().getValue();
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public E getLast()
     {
         return getLastNode().getValue();
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public E peekFirst()
     {
         return isEmpty() ? null : getFirst();
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public E peekLast()
     {
         return isEmpty() ? null : getLast();
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public boolean removeFirstOccurrence(Object o)
     {
         ListNode<E> node = nodeOf(o);
@@ -798,10 +745,6 @@ public class DoublyLinkedList<E>
         return false;
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public boolean removeLastOccurrence(Object o)
     {
         ListNode<E> node = lastNodeOf(o);
@@ -814,46 +757,26 @@ public class DoublyLinkedList<E>
 
     // Queue methods
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public boolean offer(E e)
     {
         return offerLast(e);
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public E remove()
     {
         return removeFirst();
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public E poll()
     {
         return pollFirst();
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public E element()
     {
         return getFirst();
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public E peek()
     {
         return peekFirst();
@@ -861,19 +784,11 @@ public class DoublyLinkedList<E>
 
     // Stack methods
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public void push(E e)
     {
         addFirst(e);
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public E pop()
     {
         return removeFirst();
@@ -1003,10 +918,6 @@ public class DoublyLinkedList<E>
         return reverseIterator(new ListNodeIteratorImpl(size, startNode.next));
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
     public NodeIterator<E> descendingIterator()
     {
         return reverseIterator(listIterator(size));

